applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_adjoint_solver.py

```diff
@@ -41,6 +50,8 @@ def __init__(self, model, custom_settings):
         # Construct the base solver.
         super(PotentialFlowAdjointSolver, self).__init__(model, custom_settings)
 
+        gradient_mode = self.response_function_settings["gradient_mode"].GetString()
+        self.settings["formulation"].AddEmptyValue("gradient_mode").SetString(gradient_mode)
```

Comment:
Can we just define this inside the formulation settings?

```
       "formulation"  : {
            "element_type": "incompressible",
            "gradient_mode": "semi_analytic"

        },

```
I think you dont need to do the AddEmptyValue here.

Reply:
Sorry I don't understand what you mean. Do you mean that we should not be defining the gradient_mode in the formulation settings or that we should include it but that instead of adding an empty value "manually", we should include it from the very beginning?

The thing is that I was thinking how to select the adjoint element: semi_analytic or analytic. Since there is already a gradient mode setting in the response_function_settings I didn't want to duplicate this setting. Therefore I chose to read the gradient mode from the response function settings and add it to the formulation settings. Probably there is a better way of doing it...